Automattic/pocket-casts-ios
# [Deselect Chapters] Purchase flow

| 📘 Part of: #1396 |
|:---:|

Adds the purchase flow to the Deselection chapters feature.

https://github.com/Automattic/pocket-casts-ios/assets/7040243/fad4d9bb-0e25-4202-b011-49deeca70042

These are podcasts with chapters you can use to test:

* Clublife: https://pca.st/rLzZCv
* ATP: https://pca.st/accidentaltech

## To test

3. Hit <kbd>Command</kbd>+<kbd>Shift</kbd>+<kbd>,</kbd> to open the Scheme editor
4. Click the options tab
5. Select the `Pocket Casts Configuration.storekit` file in the StoreKit configuration dropdown
2. Run the app 
3. Go to Profile > Settings > Beta Features > `deselectChapters` and turn it on
4. Login into an account with Plus
5. Play any episode with chapters
6. Go to the Chapters tab
7. ✅ You should see the Patron heart
8. Tap "Skip chapters"
9. ✅ You should see the Patron screen upgrade
10. Purchase it and dismiss the screen
11. ✅ The chapter selection should be displayed, the button is now "Done" and there's no icon
12. Close and reopen the player and go to Chapters
13. ✅ There should be no icon and you can toggle chapter selection
14. Go to Profile > Settings > Developer > Set to No Plus
15. Go to `PaidFeature.swift` and change `deselectChapters` to `.plusFeature`
16. ✅ Repeat the testing steps, this time you should see the Plus upgrade prompt and the Plus icon

## Checklist
